File: examples/has_duplicate.py
# Demo 1 — the classic O(n^2) -> O(n) rewrite.
# Compares every pair of elements to find a repeat. Correct, but quadratic.
# Expected: SBLLM replaces the nested loop with a single pass using a set.

def has_duplicate(numbers):
    for i in range(len(numbers)):
        for j in range(len(numbers)):
            if i != j and numbers[i] == numbers[j]:
                return True
    return False

File: examples/has_duplicate.cpp
// C++ demo — the same quadratic duplicate check, compiled with -std=c++17 -O3
// (the paper's own settings). Expected: SBLLM replaces the nested loop with a
// single pass using an unordered_set.

#include <vector>

bool has_duplicate(std::vector<int> numbers) {
    for (size_t i = 0; i < numbers.size(); ++i) {
        for (size_t j = 0; j < numbers.size(); ++j) {
            if (i != j && numbers[i] == numbers[j]) return true;
        }
    }
    return false;
}
